pkg/features/feature.go

```diff
@@ -3,21 +3,36 @@ package features
 import (
 	"os"
 	"strings"
+
+	"github.com/stackrox/rox/pkg/buildinfo"
 )
 
-type stage bool
+type mode int
 
+// mode bits:
 const (
-	DevPreview  = false
-	TechPreview = true
+	releaseStageBit mode = 1 << iota
+	defaultValueBit
+	unchangeableBit
+)
+
+const (
+	// Const integer 1 for release, 0 for development build:
+	releaseBuild int = 1 - (len(buildinfo.BuildFlavor)-len("release"))/4
+
+	// mode values:
+	devPreview         mode = 0 * releaseStageBit
```

Comment:
Why not use the option pattern here? I think that would be less complicated and works better when modes are orthogonal like `devPreview`/`techPreview` and `enabled`/`disabled`.

Reply:
Dunno, for me the bit mask works just fine for orthogonal boolean values:

```go
ScannerV4Support = registerFeature(enabled|unchangeableInProd, "Enable Scanner V4 Support", "ROX_SCANNER_V4_SUPPORT")
// vs
ScannerV4Support = registerFeature("Enable Scanner V4 Support", "ROX_SCANNER_V4_SUPPORT", withEnabled(true), withUnchangeableInProd(true))
```

Reply:
See #12101 with the Options implementation.